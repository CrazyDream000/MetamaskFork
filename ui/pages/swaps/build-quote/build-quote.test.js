import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import {
  renderWithProvider,
  createSwapsMockStore,
  setBackgroundConnection,
} from '../../../../test/jest';
import BuildQuote from '.';

const middleware = [thunk];
const createProps = (customProps = {}) => {
  return {
    inputValue: '5',
    onInputChange: jest.fn(),
    ethBalance: '6 ETH',
    setMaxSlippage: jest.fn(),
    maxSlippage: 15,
    selectedAccountAddress: 'selectedAccountAddress',
    isFeatureFlagLoaded: false,
    shuffledTokensList: [],
    ...customProps,
  };
};

setBackgroundConnection({
  resetPostFetchState: jest.fn(),
});

describe('BuildQuote', () => {
  it('renders the component with initial props', () => {
    const store = configureMockStore(middleware)(createSwapsMockStore());
    const props = createProps();
    const { getByText } = renderWithProvider(<BuildQuote {...props} />, store);
    expect(getByText('Swap from')).toBeInTheDocument();
    expect(getByText('Swap to')).toBeInTheDocument();
    expect(getByText('ETH')).toBeInTheDocument();
    expect(getByText('Slippage Tolerance')).toBeInTheDocument();
    expect(getByText('2%')).toBeInTheDocument();
    expect(getByText('3%')).toBeInTheDocument();
    expect(getByText('Review Swap')).toBeInTheDocument();
    expect(
      document.querySelector('.slippage-buttons__button-group'),
    ).toMatchSnapshot();
  });
});
