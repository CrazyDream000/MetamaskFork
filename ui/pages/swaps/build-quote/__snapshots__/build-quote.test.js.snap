// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildQuote renders the component with initial props 1`] = `
<div
  class="button-group slippage-buttons__button-group radio-button-group"
  role="radiogroup"
>
  <button
    aria-checked="false"
    class="button-group__button radio-button"
    data-testid="button-group__button0"
    role="radio"
  >
    2%
  </button>
  <button
    aria-checked="false"
    class="button-group__button radio-button"
    data-testid="button-group__button1"
    role="radio"
  >
    3%
  </button>
  <button
    aria-checked="true"
    class="button-group__button slippage-buttons__button-group-custom-button radio-button--danger radio-button button-group__button--active radio-button--active"
    data-testid="button-group__button2"
    role="radio"
  >
    15
    <div
      class="slippage-buttons__percentage-suffix"
    >
      %
    </div>
  </button>
</div>
`;
